resources/vep/plugins/Capice.pm

```diff
@@ -0,0 +1,146 @@
+sub new {
+    if (!(defined $self)) {
+        my $class = shift;
+        $self = $class->SUPER::new(@_);
+        my $file = $self->params->[0];
+        die("ERROR: input file not specified\n") unless $file;
+        parse_file($file);
+    }
+    return $self;
+}
+
+sub create_key {
+    my $chr = $_[0];
+    my $pos = $_[1];
+    my $ref = $_[2];
+    my $alt = $_[3];
+    my $gene = $_[4];
+    my $transcript = $_[5];
+    return "${chr}_${pos}_${ref}_${alt}_${gene}_${transcript}";
+}
+
+sub parse_file_header {
+    my @tokens = split /\t/, $_[0];
+
+    my $col_idx;
+    for my $i (0 .. $#tokens) {
+        if ($tokens[$i] eq "chr") {
+            $col_idx->{idx_chr} = $i;
+        }
+        if ($tokens[$i] eq "pos") {
+            $col_idx->{idx_pos} = $i;
+        }
+        if ($tokens[$i] eq "ref") {
+            $col_idx->{idx_ref} = $i;
+        }
+        if ($tokens[$i] eq "alt") {
+            $col_idx->{idx_alt} = $i;
+        }
+        if ($tokens[$i] eq "transcript") {
+            $col_idx->{idx_transcript} = $i;
+        }
+        if ($tokens[$i] eq "gene_id") {
+            $col_idx->{idx_gene} = $i;
+        }
+        if ($tokens[$i] eq "id_source") {
+            $col_idx->{idx_source} = $i;
+        }
+        if ($tokens[$i] eq "score") {
+            $col_idx->{idx_score} = $i;
+        }
+        if ($tokens[$i] eq "suggested_class") {
+            $col_idx->{idx_class} = $i;
+        }
+    }
+    return $col_idx;
+}
+
+sub parse_file {
+    my %capice_map;
+    open(FH, "gunzip -c @_ |") or die $!;
+
+    chomp(my $header = <FH>);
+    $header =~ s/\s*\z//;
+    my $col_idx = parse_file_header($header);
+
+    while (my $line = <FH>) {
+        $line =~ s/\s*\z//;
+        my @tokens = split /\t/, $line;
+
+        my $key = create_key($tokens[$col_idx->{idx_chr}], $tokens[$col_idx->{idx_pos}], $tokens[$col_idx->{idx_ref}], $tokens[$col_idx->{idx_alt}], $tokens[$col_idx->{idx_gene}], $tokens[$col_idx->{idx_transcript}]);
+
+        my %values;
+        $values{score} = $tokens[$col_idx->{idx_score}];
+        $values{classification} = $tokens[$col_idx->{idx_class}];
+        $capice_map{$key} = \%values;
+    }
+    close FH;
+
+    $self->{capice_map} = \%capice_map;
+}
+
+sub run {
+    my ($self, $tva) = @_;
+
+    my $bvfoa = $tva->base_variation_feature;
+    my @vcf_line = @{$bvfoa->{_line}};
+    my $chr = $vcf_line[0];
+    my $pos = $vcf_line[1];
+    my $ref = $vcf_line[3];
+    my $alt = $vcf_line[4];
+
+    return {} unless ($tva->transcript->{_gene_symbol_source} eq "EntrezGene");
+
+    my $gene = $tva->transcript->{_gene_stable_id};
+    my $transcript_id = $tva->transcript->{stable_id};
+    my $key = create_key($chr, $pos, $ref, $alt,$gene,$transcript_id);
+
+    my $result = ();
+    $result->{CAPICE_SC} = undef;
+    $result->{CAPICE_CL} = undef;
+    my $value = $self->{capice_map}{$key};
+    if($value) {
+        my %value_map = %{$value};
+        $result->{CAPICE_SC} = $value_map{score};
```

Comment:
is the value of score handled like string a numeric characters or does Perl treat it as a floating point value? in the latter case I am wondering whether values are treated correctly.